### ISSUE TITLE
Fix source-gen issue with a binding member access expression being on a separate line from the member expression

### ISSUE THREAD

**Issue by layomia** [dotnet/runtime#90851]
**Same line**

Interceptor location is successfully parsed. Project compiles:

```cs
var optionsBuilder = services.AddOptions<AppSettings>().BindConfiguration(nameof(AppSettings));

var optionsBuilder = services
    .AddOptions<AppSettings>().BindConfiguration(nameof(AppSettings));
```

**Different line**

Parsed interceptor location is inaccurate. Compilation fails with a CSC interceptors error stating that the generated `[InterceptsLocation]` points to an invalid location.

```cs
var optionsBuilder = services.AddOptions<AppSettings>()
    .BindConfiguration(nameof(AppSettings));

var optionsBuilder = services
    .AddOptions<AppSettings>()
    .BindConfiguration(nameof(AppSettings));
```

**Comment by layomia** [comment]
@captainsafia @eerhardt this will affect the stage 2 showcase [app](https://github.com/aspnet/Benchmarks/blob/main/src/BenchmarksApps/TodosApi/AppSettings.cs#L36-L37).

FYI @ericstj.

**Comment by layomia** [comment]
Fixed for 8.0 in https://github.com/dotnet/runtime/pull/91218.
